server.go

```diff
@@ -47,9 +46,9 @@ var (
 // RunServer runs server for exporter with given name (it is used on landing page) on given address,
 // exposing metrics under given path.
 // Function never returns.
-func RunServer(name, addr, path string, errorHandling promhttp.HandlerErrorHandling) {
+func RunServer(name, addr, path string, handler func(http.ResponseWriter, *http.Request)) {
```

Comment:
Accept `http.Handler` interface just like `runHTTP` and `runHTTPS`. That way anything implementing that interface can be passed, including `http.HandlerFunc` and any other specific type implementing `http.Handler`.